xapian-applications/omega/date.cc

```diff
@@ -144,7 +147,40 @@ date_range_filter(const string & date_start, const string & date_end,
 {
     int y1, m1, d1, y2, m2, d2;
     if (!date_span.empty()) {
-	time_t secs = atoi(date_span.c_str()) * (24 * 60 * 60);
+    	time_t secs;
+	// For x32 system use long (32bit)
+	// For x64 system use long long (64bit)
+	if (sizeof(int *) < 8) {
+	    long str_to_long;
+	    errno = 0;
+	    str_to_long = strtol(date_span.c_str(),NULL,0);
+	    if (errno == ERANGE) {
+		switch(str_to_long) {
+		    case LONG_MIN:
+			throw (string("Date span underflow: ") + string(strerror(errno)));
+		    break;
```

Comment:
`break;` is mis-indented here, but there's no point having a `break;` after a throw anyway.